### PR TITLE
Using cmake for building the plugin

### PR DESCRIPTION
**Pull request by razdoburdin**
1. This PR greatly simplify building of sycl plugin.

Instead of patching with the following using of external script, user just needs to call cmake with `DPLUGIN_UPDATER_ONEAPI=ON `

The full building procedure looks like this now:
```
mkdir _build; cd _build;
cmake -DPLUGIN_UPDATER_ONEAPI=ON ..
make
cd ../python-package
pip install .
```

For building with google tests support:
`cmake -DGOOGLE_TEST=ON -DUSE_DMLC_GTEST=ON -DPLUGIN_UPDATER_ONEAPI=ON ..`

For launching google tests:
`./testxgboost`

For launching only sycl related cpp tests:
`./testxgboost --gtest_filter=Sycl*`

2. Fixing the found bugs.

### CODE REVIEW

**Comment by ahuber21** [comment]
@razdoburdin could you provide a second PR that fixes the pylint errors in `xgboost/spark/estimator.py`?

**Comment by razdoburdin** [comment]
> @razdoburdin could you provide a second PR that fixes the pylint errors in `xgboost/spark/estimator.py`?

It is the XGBoost CI problem. Some checks doesn't work popper in case PR is made not to the main repo.

**Comment by razdoburdin** [comment]
> Should we keep updater_oneapi name? It worth renaming plugin name as well?

@napetrov
Yes, we will rename plugin and all related files/variables/classes. But not in the current PR.

File: plugin/CMakeLists.txt [hunk 1]
```diff
@@ -24,7 +25,8 @@ if (PLUGIN_UPDATER_ONEAPI)
     POSITION_INDEPENDENT_CODE ON)
   if (USE_OPENMP)
     find_package(OpenMP REQUIRED)
-    target_link_libraries(oneapi_plugin PUBLIC OpenMP::OpenMP_CXX)
+    set_target_properties(oneapi_plugin PROPERTIES
+    COMPILE_FLAGS "-fsycl -qopenmp")
```
[the remark this answers is not in the document]
Comment (razdoburdin) [reply plugin/CMakeLists.txt:29]: > Do we need openmp flag? 
Unfortunately yes. I didn't yet fully cleaned the code from openmp in the plugin.

> From deps perspective I think we should use icx just for plugging and sycl kernels?
Currently we use `icpx `for compilation of the sycl-plugin and for linking. For the main part of xgboost the system defined compiler (the $CXX) is used.

**Pull request merged**